### PR TITLE
GDScript: Disable LSP if either jsonrpc or websocket are disabled

### PR DESCRIPTION
**Pull request by akien-mga**
Follow-up to #29780, solves this comment: https://github.com/godotengine/godot/pull/29780#discussion_r318520008

CC @Geequlim @ankitpriyarup

### CODE REVIEW

**Comment by akien-mga** [comment]
Wait that's likely not enough as things get registered in the base module too.

**Comment by akien-mga** [comment]
Should be ready for review.

File: modules/gdscript/SCsub [hunk 1]
```diff
@@ -8,5 +8,12 @@ env_gdscript = env_modules.Clone()
 env_gdscript.add_source_files(env.modules_sources, "*.cpp")
 
 if env['tools']:
-	env_gdscript.add_source_files(env.modules_sources, "./editor/*.cpp")
-	env_gdscript.add_source_files(env.modules_sources, "./language_server/*.cpp")
+    env_gdscript.add_source_files(env.modules_sources, "./editor/*.cpp")
+
+    # Those two modules are required for the language server protocol
+    if env['module_jsonrpc_enabled'] and env['module_websocket_enabled']:
```
Comment (Geequlim) [inline modules/gdscript/SCsub]: Does the `module_jsonrpc_enabled` and `module_websocket_enabled` are reliable here ?
Does the order of SCsub runing matter? Is it possiable the `SCsub` of the GDScript mulde run before jsonrpc or websocket

Comment (akien-mga) [reply modules/gdscript/SCsub]: I think so, in `SConstruct` the `module_*_enabled` as defined for each module at line 185, while the `modules/SCsub` is read at line 517.
But that's a good comment, we should be careful around those.

Comment (akien-mga) [reply modules/gdscript/SCsub]: It's used in a few other places this way and seems to work so far:
```
modules/webrtc/SCsub
9:use_gdnative = env_webrtc["module_gdnative_enabled"]

core/crypto/SCsub
8:has_module = env["module_mbedtls_enabled"]

modules/gdnative/net/SCsub
8:has_webrtc = env_net["module_webrtc_enabled"]
```

**Pull request merged**